File: backend-java/src/main/java/com/dramaflow/model/WatchRecord.java
package com.dramaflow.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "watch_records",
       uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "episode_id"}))
public class WatchRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "episode_id", nullable = false)
    private Integer episodeId;

    private Double progress = 0.0;

    @Column(name = "last_position")
    private Double lastPosition = 0.0;

    private Boolean completed = false;

    @Column(updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    private LocalDateTime updatedAt = LocalDateTime.now();

    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }
    public Integer getUserId() { return userId; }
    public void setUserId(Integer userId) { this.userId = userId; }
    public Integer getEpisodeId() { return episodeId; }
    public void setEpisodeId(Integer episodeId) { this.episodeId = episodeId; }
    public Double getProgress() { return progress; }
    public void setProgress(Double progress) { this.progress = progress; }
    public Double getLastPosition() { return lastPosition; }
    public void setLastPosition(Double lastPosition) { this.lastPosition = lastPosition; }
    public Boolean getCompleted() { return completed; }
    public void setCompleted(Boolean completed) { this.completed = completed; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }

    @PreUpdate
    public void onUpdate() { this.updatedAt = LocalDateTime.now(); }
}

File: backend/app/models/watch_record.py
from sqlalchemy import Column, Integer, Float, Boolean, ForeignKey, DateTime, func, UniqueConstraint
from sqlalchemy.orm import relationship
from app.db.database import Base


class WatchRecord(Base):
    __tablename__ = "watch_records"

    __table_args__ = (
        UniqueConstraint("user_id", "episode_id", name="uq_user_episode"),
    )

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    episode_id = Column(Integer, ForeignKey("episodes.id"), nullable=False)
    progress = Column(Float, default=0.0)       # 0-100
    last_position = Column(Float, default=0.0)  # seconds
    completed = Column(Boolean, default=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    user = relationship("User")
    episode = relationship("Episode", back_populates="watch_records")
